main.go

```diff
@@ -1,13 +1,63 @@
 package main
 
 import (
+	"fmt"
+
+	"github.com/alecthomas/kong"
+	"github.com/sirupsen/logrus"
+
 	"github.com/Percona-Lab/mnogo_exporter/exporter"
 )
 
+var (
+	version   string
+	commit    string //nolint
+	buildDate string //nolint
+)
+
+// GlobalFlags has command line flags to configure the exporter.
+type GlobalFlags struct {
+	DSN     string `name:"mongodb.dsn" help:"MongoDB connection URI" placeholder:"mongodb://user:pass@127.0.0.1:27017/admin?ssl=true"`
+	Debug   bool   `name:"debug" short:"D" help:"Enable debug mode"`
+	Version bool   `name:"version" help:"Show version and exit"`
+}
+
 func main() {
-	e, err := exporter.New(nil)
+	var opts GlobalFlags
+	_ = kong.Parse(&opts,
+		kong.Name("mnogo_exporter"),
+		kong.Description("MongoDB Prometheus exporter"),
+		kong.UsageOnError(),
+		kong.ConfigureHelp(kong.HelpOptions{
+			Compact: true,
+		}),
+		kong.Vars{
+			"version": version,
+		})
+
+	if opts.Version {
+		fmt.Println("mnogo-exporter - MongoDB Prometheus exporter")
+		fmt.Printf("Version: %s\n", version)
+		fmt.Printf("Commit: %s\n", commit)
+		fmt.Printf("Build date: %s\n", buildDate)
+
+		return
+	}
+
+	log := logrus.New()
+
+	if opts.Debug {
+		logrus.SetLevel(logrus.DebugLevel)
+	}
+
+	exporterOpts := &exporter.Opts{
+		DSN: opts.DSN,
+	}
+
+	e, err := exporter.New(exporterOpts)
 	if err != nil {
-		panic(err)
+		log.Fatal(err)
 	}
+
```

Comment:
❓ Ok, and?.. The intention was to build an actual exporter – without any MongoDB metrics / collectors, but with default metrics provided by the Prometheus library (go process metrics, a number of scrapes, etc.) on `/metrics` path. That would also show how the connection will be established, and whatever it will be shared or not.

Reply:
It looks like part of it is done at #3. I think we should move it there – not the actual collector, but code that starts exporter and gathers data from the future collectors.